playbooks/spark-unit-test-arm64/run.yaml

```diff
@@ -0,0 +1,73 @@
+- hosts: all
+  become: yes
+  roles:
+    - install-openjdk
+    - install-maven
+    - config-hostname
+
+- hosts: all
+  tasks:
+    - name: Build and run unit tests for apache spark
+      shell:
+        cmd: |
+          set -ex
+          sudo apt-get update -y
+
+          # git snappy-1.0.5
+          wget http://repository.timesys.com/buildsources/s/snappy/snappy-1.0.5/snappy-1.0.5.tar.gz
+          tar -xvf snappy-1.0.5.tar.gz
+
+          git clone http://github.com/huangtianhua/leveldbjni
+          git clone http://github.com/huangtianhua/leveldb
+          export SNAPPY_HOME=`cd snappy-1.0.5; pwd`
+          export LEVELDBJNI_HOME=`cd leveldbjni; pwd`
+          export LEVELDB_HOME=`cd leveldb; pwd`
+
+          cd ${SNAPPY_HOME}
+          # get the latest config.guess and config.sub
+          wget -O config.guess "git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD"
+          wget -O config.sub "git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD"
+          ./configure --disable-shared --with-pic
+          make
+
+          cd ${LEVELDB_HOME}
+          export LIBRARY_PATH=${SNAPPY_HOME}
+          export C_INCLUDE_PATH=${LIBRARY_PATH}
+          export CPLUS_INCLUDE_PATH=${LIBRARY_PATH}
+          git apply ../leveldbjni/leveldb_aarch64.patch
+          make libleveldb.a
+
+          # fix the problem: 'aclocal-1.14' is missing on your system
+          cd ../
+          sudo apt-get install autoconf -y
+          wget http://ftp.gnu.org/gnu/automake/automake-1.14.tar.gz
+          tar -zxvf automake-1.14.tar.gz
+          cd automake-1.14
+          ./bootstrap.sh
+          ./configure
+          sudo make
+          sudo make install
+
+          cd ${LEVELDBJNI_HOME}
+          # fix error: configure.ac:36: error: required file 'autotools/compile' not found
+          mkdir -p ${LEVELDBJNI_HOME}/leveldbjni-linux64-aarch64/target/native-build/autotools/
+          cp /usr/local/share/automake-1.14/compile ${LEVELDBJNI_HOME}/leveldbjni-linux64-aarch64/target/native-build/autotools/
+          # build leveldbjni-linux64-aarch64
+          mvn install -P download -P linux64-aarch64
+          # build leveldbjni-all
+          mvn install -P download -P all
+
+          # use local levedbjni jar built above
+          export SPARK_HOME='/home/zuul/src/github.com/theopenlab/spark/'
```

Comment:
please do not use explicit `/home/zuul/....` directory, use ansible variable instead

Reply:
Done